processor/src/main/kotlin/permissions/dispatcher/processor/util/Extensions.kt

```diff
@@ -7,15 +7,20 @@ import permissions.dispatcher.OnShowRationale
 import permissions.dispatcher.processor.TYPE_UTILS
 import javax.lang.model.element.Element
 import javax.lang.model.element.ExecutableElement
+import javax.lang.model.element.PackageElement
 import javax.lang.model.element.TypeElement
 import javax.lang.model.type.TypeMirror
 
 /**
  * Returns the package name of a TypeElement.
  */
 fun TypeElement.packageName(): String {
-    val qn = this.qualifiedName.toString()
-    return qn.substring(0, qn.lastIndexOf('.'))
+    val enclosingElement = enclosingElement
+    return when (enclosingElement) {
+        is TypeElement -> enclosingElement.packageName()
+        is PackageElement -> enclosingElement.qualifiedName.toString()
+        else -> ""
```

Comment:
In these cases, would it be feasible to walk up the "enclosingElement" hierarchy to check for `TypeElement`/ `PackageElement` higher up the chain?